lib/Controller/MessagesController.php

```diff
@@ -716,28 +733,40 @@ public function removeTag(int $id, string $imapLabel): JSONResponse {
 	 *
 	 * @param int $accountId
 	 * @param string $folderId
-	 * @param int $id
+	 * @param array $ids
 	 *
 	 * @return JSONResponse
 	 *
 	 * @throws ClientException
 	 * @throws ServiceException
 	 */
-	public function destroy(int $id): JSONResponse {
+	public function deleteMessages(array $ids): JSONResponse {
+		$uids = [];
 		try {
-			$message = $this->mailManager->getMessage($this->currentUserId, $id);
-			$mailbox = $this->mailManager->getMailbox($this->currentUserId, $message->getMailboxId());
-			$account = $this->accountService->find($this->currentUserId, $mailbox->getAccountId());
-		} catch (DoesNotExistException $e) {
+			// Makes sure all messages to be deleted come from the same mailbox
+			$message = $this->mailManager->getMessage($this->currentUserId, $ids[0]);
+			$firstMailbox = $this->mailManager->getMailbox($this->currentUserId, $message->getMailboxId());
+			$firstAccount = $this->accountService->find($this->currentUserId, $firstMailbox->getAccountId());
+			foreach ($ids as $id) {
+				$message = $this->mailManager->getMessage($this->currentUserId, $id);
```

Comment:
I think we could use `MessageMapper.findByUids` here to fetch all messages in one query. In addition we could sent the mailboxId from the frontend to skip getMessage for the first message to find the right mailboxId.

```
$firstMailbox = $this->mailManager->getMailbox($this->currentUserId, $_POST['mailboxId']);
if ($firstMailbox === null) {
  NO!
}
$messages = messageMapper.findByUids($firstMailbox, $ids);
```

Reply:
This would silently ignore messages from other mailboxes though (use case is: user selects several messages, from several mailboxes, *from within the priority inbox*. With your change, the controller would silently ignore those messages that don't come from $firstMailbox)

But, then, the frontend may provide a batch like this to solved this issue:
```
batch = {
   $mailboxId => {
        $messageId_1,
        $messageId_2,
   },
  $mailboxId_2 => {
        $messageId_3,
        $messageId_4,
        ...
   },
   ...
}
```

Reply:
True. I forgot the priority inbox. So we need a method to find all messages by messageId/currentUserId and another method to find multiple mailboxes by mailboxId/currentUserId.